File: adventday1/go/main.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type item struct{ calorie int }
type elf struct {
	inventory     []item
	totalCalories int
}
type Elves []elf

func main() {
	ledger, _ := os.ReadFile("../input.txt")
	elfLedger := strings.Split(string(ledger), "\n\n")

	var best, second, third elf
	var elves Elves
	for _, elfInventory := range elfLedger {
		elf := &elf{}
		eInventory := strings.Split(elfInventory, "\n")
		for _, eItem := range eInventory {
			inventoryItem, _ := strconv.Atoi(eItem)
			elf.totalCalories += inventoryItem
			elf.inventory = append(elf.inventory, item{calorie: inventoryItem})
		}

		elves = append(elves, *elf)
	}

	sort.Slice(elves, func(i, j int) bool {
		return elves[j].totalCalories < elves[i].totalCalories
	})

	for _, elf := range elves {
		switch {
		case elf.totalCalories > best.totalCalories:
			best = elf
		case elf.totalCalories < best.totalCalories && elf.totalCalories > second.totalCalories:
			second = elf
		case elf.totalCalories < second.totalCalories && elf.totalCalories > third.totalCalories:
			third = elf
		}
	}

	fmt.Printf("best: %v\n", best.totalCalories)
	fmt.Printf("second: %v\n", second.totalCalories)
	fmt.Printf("third: %v\n", third.totalCalories)
	fmt.Printf("Top 3: %v", best.totalCalories+second.totalCalories+third.totalCalories)
}

File: adventday1/rust/main.rs
use std::fs;

fn main() {
    let ledger = fs::read_to_string("../input.txt").unwrap();
    let mut elves = ledger
        .split("\n\n")
        .map(|elf| {
            elf.split('\n')
            .map(|item| str::parse::<u64>(item).unwrap_or(0))
            .sum()
        })
        .collect::<Vec<u64>>();

    elves.sort_by(|a, b| b.cmp(a));

    println!("best: {:?}", elves[0]);
    println!("second: {:?}", elves[1]);
    println!("third: {:?}", elves[2]);
    println!("Top 3: {}", elves[0] + elves[1] + elves[2]);
}